lib/lz4frame.c

```diff
@@ -345,32 +342,37 @@ size_t LZ4F_compressFrame_usingCDict(void* dstBuffer, size_t dstCapacity,
     if (srcSize <= LZ4F_getBlockSize(prefs.frameInfo.blockSizeID))
         prefs.frameInfo.blockMode = LZ4F_blockIndependent;   /* only one block => no need for inter-block link */
 
-    if (prefs.compressionLevel < LZ4HC_CLEVEL_MIN) {
-        cctxI.lz4CtxPtr = &lz4ctx;
-        cctxI.lz4CtxLevel = 1;
-    }  /* fast compression context pre-created on stack */
+    if (cctx == NULL) {
```

Comment:
I believe this could be done differently : 
make sure `LZ4F_compressFrame()` provides a valid context to `LZ4F_compressFrame_usingCDict()`.